test/common/quic/envoy_quic_server_session_test.cc

```diff
@@ -408,6 +410,55 @@ TEST_P(EnvoyQuicServerSessionTest, OnResetFrame) {
                     ->value());
 }
 
+TEST_P(EnvoyQuicServerSessionTest, OnResetFrameIetfQuic) {
+  installReadFilter();
+  if (!quic::VersionUsesHttp3(quic_version_[0].transport_version)) {
+    return;
+  }
+  Http::MockRequestDecoder request_decoder;
+  Http::MockStreamCallbacks stream_callbacks;
+  auto stream1 =
+      dynamic_cast<EnvoyQuicServerStream*>(createNewStream(request_decoder, stream_callbacks));
+  // Receiving RESET_STREAM alone should only close read side.
+  quic::QuicRstStreamFrame rst1(/*control_frame_id=*/1u, stream1->id(),
+                                quic::QUIC_ERROR_PROCESSING_STREAM, /*bytes_written=*/0u);
+  envoy_quic_session_.OnRstStream(rst1);
+  Http::TestResponseHeaderMapImpl response_headers{{":status", "200"}};
+  stream1->encodeHeaders(response_headers, true);
+
+  EXPECT_EQ(1U, TestUtility::findCounter(
+                    static_cast<Stats::IsolatedStoreImpl&>(listener_config_.listenerScope()),
+                    "http3.downstream.rx.quic_reset_stream_error_code_QUIC_ERROR_PROCESSING_STREAM")
+                    ->value());
+
+  EXPECT_CALL(http_connection_callbacks_, newStream(_, false))
+      .WillOnce(Invoke([&request_decoder, &stream_callbacks](Http::ResponseEncoder& encoder,
+                                                             bool) -> Http::RequestDecoder& {
+        encoder.getStream().addCallbacks(stream_callbacks);
+        return request_decoder;
+      }));
+  quic::QuicStream* stream2 = envoy_quic_session_.GetOrCreateStream(stream1->id() + 4u);
+  quic::QuicRstStreamFrame rst2(/*control_frame_id=*/1u, stream2->id(), quic::QUIC_REFUSED_STREAM,
+                                /*bytes_written=*/0u);
+  quic::QuicStopSendingFrame stop_sending(/*control_frame_id=*/1u, stream2->id(),
+                                          quic::QUIC_REFUSED_STREAM);
+  // Receiving both STOP_SENDING and RESET_STREAM should close the stream.
+  EXPECT_CALL(stream_callbacks,
+              onResetStream(Http::StreamResetReason::RemoteRefusedStreamReset, _));
+  EXPECT_CALL(*quic_connection_, SendControlFrame(_))
+      .WillOnce(Invoke([stream_id = stream2->id()](const quic::QuicFrame& frame) {
+        EXPECT_EQ(stream_id, frame.rst_stream_frame->stream_id);
+        EXPECT_EQ(quic::QUIC_REFUSED_STREAM, frame.rst_stream_frame->error_code);
+        return false;
+      }));
+  envoy_quic_session_.OnStopSendingFrame(stop_sending);
+  envoy_quic_session_.OnRstStream(rst2);
```

Comment:
Should you test both orderings: stop before rst and rst before stop?  I think that the end state after both should be the same.

Reply:
done